File: strivers_a2z/dp/70climbingStairs.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int climbStairs(int n) {
        int a = 1;
        int b = 1;
        int ans = 1;
        for(int i=2;i<=n;i++){
            ans = a+b;
            a = b;
            b = ans;
        }
        return ans;
    }
};

int main(){
    return 0;
}

File: NeetCode150/70climbingStairs.py
# https://leetcode.com/problems/climbing-stairs/

class Solution:
    def climbStairs(self, n: int) -> int:
        dp = []*n
        dp[n-1] = 1
        dp[n-2] = 2
        for i in range(n-3,0,-1):
            dp[i] =  dp[i+1] + dp[i+2]

        return dp[0]